### PR TITLE
Add mi event type code list

### LINKED ISSUE #74: Add MI_EventTypeCode
This is the code list
(removed)

### PR DESCRIPTION
**Pull request by J-Oliveros**
Closes #74 

# Changes

## New Code List

MI_EventTypeCode

### Testing
https://mdtools.jwaspin.dev/

### CODE REVIEW

**Comment by dwalt** [comment]
@jwaspin Yes, this is the ordering in ISO 19115-2: Amd 2022.

1. announcement
2. calibration
3. calibrationCoefficientUpdate
4. dataLoss
5. fatal
6. manoeuvre
7. missingData
8. notice
9. prelaunch
10. severe
11. switchOff
12. switchOn
13. clean

File: resources/iso_eventTypeCode.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  - {code: "fatal", codeName: "fatal", description: "event that renders the platform/instrument/sensor unusable"}
+  - {code: "manoeuvre", codeName: "manoeuvre", description: "event related to a manoeuvre of the platform/instrument/sensor"}
+  - {code: "missingData", codeName: "missingData", description: "event related to missing data from the platform/instrument/sensor"}
+  - {code: "notice", codeName: "notice", description: "notice about events related to the platform/instrument/sensor"}
+  - {code: "prelaunch", codeName: "prelaunch", description: "event related to prelaunch period for the platform/instrument/sensor"}
+  - {code: "severe", codeName: "severe", description: "event with significant impact on the performance of the platform/instrument/sensor"}
+  - {code: "switchOff", codeName: "switchOff", description: "event related to switching off platform/instrument/sensor"}
+  - {code: "switchOn", codeName: "switchOn", description: "event related to switching on platform/instrument/sensor"}
+  - {code: "clean", codeName: "clean", description: "event related to cleaning the platform/instrument/sensor"}
```
Comment (jwaspin) [inline resources/iso_eventTypeCode.yml:24]: Change to numerical `code` values using the ordering in the comments

**Review by jwaspin — CHANGES_REQUESTED** [review]

**Review by jwaspin — APPROVED** [review]

File: resources/iso_eventTypeCode.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
+---
+# mdJson codelists
+# base codelist for ISO 19115-2, 19115-1
+
+codelistType: "staticList"
+codelistName: "iso_eventTypeCode"
+source: "ISO"
+sourceName: "MI_EventTypeCode"
+extensible: true
+description: "code indicating whether the data contained in this packet is real (originates from live-fly or other non-simulated operational sources), simulated (originates from target simulator sources), or synthesized (a mix of real and simulated data)."
```
Comment (hmaier-fws) [inline resources/iso_eventTypeCode.yml:10]: @dwalt this description doesn't seem match the codelist. Can you verify. According to https://schemas.isotc211.org/schemas/Resources/codelists.xml the definition is "type of event related to platform/instrument/sensor"

Comment (jwaspin) [reply resources/iso_eventTypeCode.yml]: @dwalt @hmaier-fws That description was not sourced from anything official. It has been changed to reflect the correct description.

**Review by hmaier-fws — APPROVED** [review]
ok

**Pull request merged**